### PR TITLE
Interpreter functional + unittest script

### PR DESCRIPTION
**Pull request by jaytaph**
Added a unittest script to detect differences between the output of a script and the expected output (as defined inside the script source)

Also implemented while and if / else if / else control structures which are working..

### CODE REVIEW

File: tests/006.sf [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 $a = $a + 1;
-print $a;       // 8
+print $a;
 
 $a = $a - 1;
-print $a;       // 7
+print $a;
+
+/* Output:
+10106786
```
Comment (Freeaqingme) [inline tests/006.sf]: I'm presuming the lsat digit should be 7 (seven)?

File: tests/sf-unittest.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#/bin/sh
+
+if [ $# -ne 1 ] ; then
+	echo "Usage: $0 <sf.name>";
+	exit;
+fi
+
+# Use colordiff if available
+diff_bin=`which colordiff`
```
Comment (Freeaqingme) [inline tests/sf-unittest.sh:9]: s/which/env ?

Comment (jaytaph) [inline tests/sf-unittest.sh:9]: nope.  which is good

File: tests/006.sf (hunk 1 above)
Comment (jaytaph) [inline tests/006.sf]: Yes, that is correct. I've changed it to 6, to check when a unit-test fails..  didn't changed it back

**Pull request merged**